crml/transaction-payment/src/mock.rs

```diff
@@ -142,10 +142,20 @@ impl BuyFeeAsset for Module<Runtime> {
 	) -> sp_std::result::Result<Self::Balance, DispatchError> {
 		if exchange_op.asset_id() == VALID_ASSET_TO_BUY_FEE {
 			if exchange_op.max_payment() == 0 {
-				return Err(DispatchError::Other("no money"));
+				return Err(DispatchError::Module {
+					index: 1,
+					error: 15,
+					message: Some("CoreToAssetPriceAboveMaxLimit"),
```

Comment:
You shouldn't need to send a string like this. You should be able to return an enum entry/a defined const. Using hard-coded string literals is not a good practice.

Reply:
For this we wanted to return the error explicitly for testing purposes I guess lol